service_thing_aggregate/src/main/scala/dao/ObjectSearchDao.scala

```diff
@@ -0,0 +1,204 @@
+import play.api.Logger
+import play.api.db.slick.{DatabaseConfigProvider, HasDatabaseConfigProvider}
+import play.api.libs.concurrent.Execution.Implicits.defaultContext
+import slick.driver.JdbcProfile
+
+import scala.concurrent.Future
+
+/**
+ * Created by jarle on 06.10.16.
+ */
+
+sealed trait FieldValue
+
+case class EmptyValue() extends FieldValue
+
+case class LiteralValue(v: String) extends FieldValue
+
+/*If v contains a value which needs to be escaped, escapeChar contains the appropriate escape character.
+If v doesn't contains a value which needs to be escaped, escapeChar =
+ */
+case class WildcardValue(v: String, escapeChar: Char) extends FieldValue
+
+//#If we (in the future) need to treat some characters as invalid: case class InvalidValue(errorMessage: String) extends FieldValue
+
+class ObjectSearchDao @Inject() (
+    val dbConfigProvider: DatabaseConfigProvider
+) extends HasDatabaseConfigProvider[JdbcProfile] {
+
+  val logger = Logger(classOf[ObjectAggregationDao])
+
+  import driver.api._
+
+  private val musitThingTable = TableQuery[MusitThingTable]
+
+  def testInsert(museumId: Int, mthing: MusitThing): Future[Long] = {
+    val dto = MusitThingDto.fromDomain(museumId, mthing)
+
+    val action = musitThingTable returning musitThingTable.map(_.id) += dto
+    db.run(action)
+  }
+
+  val noEscapeChar = '\u0000' //Needs to be the same as Slicks no-escape-char value! (Default second parameter value to the like function)
+
+  val escapeChar = '¤' //can be any char really. Public just to let the unit tests get at it
+
+  def classifyValue(rawValue: String): FieldValue = {
+    //Since we build up a Slick Query object, we don't need to verify that the rawValue is "safe",
+    // the database engine will validate the parameter value.
+
+    //So security-wise, we don't need to guard against '--' etc here.
+    //(Else we could have flagged values containing "--" etc as InvalidValue here)
+    // We use '*' as wildcard symbol. And treat both '%' and '_' as ordinary characters, both in like-tests and =-tests.
+
+    if (rawValue.isEmpty) EmptyValue()
+    else if (rawValue.contains('*')) {
+      val wValue = rawValue
+        .replace(escapeChar.toString, s"${escapeChar}${escapeChar}")
+        .replace("%", s"${escapeChar}%")
+        .replace("_", s"${escapeChar}_")
+        .replace('*', '%')
+
+      /*Note that in the above expression, order is vital!
+        It is essential that the escapeChar -> escapeChar+escapeChar is done before the replacements which
+       introduces any escapeChars and that  %->escapeChar happens before *->'%'
+       */
+
+      val esc = if (wValue.contains(escapeChar)) escapeChar else noEscapeChar
+
+      WildcardValue(wValue, esc)
+    } else {
+      LiteralValue(rawValue)
+    }
+  }
+
+  def fieldValueToMusitResult(fieldValue: FieldValue): MusitResult[FieldValue] = {
+    MusitSuccess(fieldValue)
+  }
+
+  def classifyValueAsMusitResult(rawValue: String): MusitResult[FieldValue] = {
```

Comment:
The commit is only "lost" on the master branch when merged to create a more linear history. Not on the branch it was created on. What typically happens is that the branch used is removed when no longer needed. So in that respect, the commit is "lost" on github. However, on your machine you will still have the same branch with full history.

This is one of many reasons why working on forks are consider to be a better approach. It allows you to keep the history _you_ want to keep. Isolated and separately from the upstream repository.

In production code however, only code and comments that are relevant to the execution of the system should be present. And commented out code is typically not merged in.